components/benefit_sponsors/app/models/benefit_sponsors/organizations/factories/profile_factory.rb

```diff
@@ -418,9 +421,32 @@ def persist_representative!
             end
             person.save!
             profile.update_attributes!(primary_broker_role_id: person.broker_role.id)
+            create_broker_agency_staff_role(profile.id)
             trigger_broker_application_confirmation_email(person)
           end
 
+          # Calls method to create Broker Agency Staff Role for the person.
+          #
+          # @note This method may raise an exception if the Broker Agency Staff Role is not created successfully.
+          #   Broker Agency Staff Role is created only if the Resource Registry feature :broker_role_consumer_enhancement is enabled.
+          #   Broker Agency Staff Role should be created only if the person already exists in the system and NOT for a new person.
+          #
+          # @param [Person] An instance of Person.
+          # @param [BenefitSponsors::Organizations::BrokerAgencyProfile] An instance of BrokerAgencyProfile.
+          # @return [Boolean] true if the Broker Agency Staff Role is created successfully.
+          def create_broker_agency_staff_role(bap_id)
+            return if matched_person.blank?
+            return if matched_person.user.blank?
+            return unless EnrollRegistry.feature_enabled?(:broker_role_consumer_enhancement)
+
+            matched_person.create_broker_agency_staff_role(
+              {
+                aasm_state: 'active',
+                benefit_sponsors_broker_agency_profile_id: bap_id
```

Comment:
Small issue here.  By setting the state directly as active and skipping the transition, we skip a number of pieces of code which would send invitations or notifications, as well as record some auditable data about the transition.

The most import is the transition under the broker agency staff role itself:
`event :broker_agency_accept, :after => [:record_transition, :send_invitation]`

We either need to trigger the transition or invoke these methods ourselves in order to keep both the database and the invitation/notification behaviour consistent.